Type error calling `fsspec.get_fs_token_paths` with S3 path in fsspec 0.9.0

We have some tests that started failing with the fsspec==0.9.0 and s3fs==0.6.0 releases this week. 

The code is calling `fsspec.get_fs_token_paths` on an s3 path. It fails in the async implementation:

```
self = <s3fs.core.S3FileSystem object at 0x7f43a673f220>, path = 'no_ttl'
maxdepth = 1, withdirs = True, kwargs = {}, detail = True, _ = [], dirs = []
files = []

    async def _find(self, path, maxdepth=None, withdirs=False, **kwargs):
        path = self._strip_protocol(path)
        out = dict()
        detail = kwargs.pop("detail", False)
        # async for?
        async for _, dirs, files in self._walk(path, maxdepth, detail=True, **kwargs):
            if withdirs:
>               files.update(dirs)
E               AttributeError: 'list' object has no attribute 'update'
```

It appears that `self._walk` [usually (always?) yields a list](https://github.com/intake/filesystem_spec/blob/bb7c23305ce76cb22cd10305f87a49009f363a11/fsspec/asyn.py#L267-L307) but `self._find` [tries to updated it like a dictionary](https://github.com/intake/filesystem_spec/blob/bb7c23305ce76cb22cd10305f87a49009f363a11/fsspec/asyn.py#L400-L402).

I haven't been able to come up with a reproducible example that I am able to share, but I can keep trying.

--- a/fsspec/asyn.py
+++ b/fsspec/asyn.py
@@ -274,7 +274,10 @@ class AsyncFileSystem(AbstractFileSystem):
         try:
             listing = await self._ls(path, detail=True, **kwargs)
         except (FileNotFoundError, IOError):
-            yield [], [], []
+            if detail:
+                yield path, {}, {}
+            else:
+                yield path, [], []
             return
 
         for info in listing:

--- a/fsspec/spec.py
+++ b/fsspec/spec.py
@@ -385,7 +385,9 @@ class AbstractFileSystem(up, metaclass=_Cached):
         try:
             listing = self.ls(path, detail=True, **kwargs)
         except (FileNotFoundError, IOError):
-            return [], [], []
+            if detail:
+                return path, {}, {}
+            return path, [], []
 
         for info in listing:
             # each info name must be at least [path]/part , but here

